app/views/govuk_publishing_components/components/_attachment.html.erb

```diff
@@ -46,20 +47,31 @@
     <% end %>
   <% end %>
 
-  <%= tag.h2 class: "gem-c-attachment__title" do %>
-    <%= link_to attachment.title, attachment.url,
-          class: "govuk-link",
-          target: target %>
-  <% end %>
+  <%= tag.div class: "gem-c-attachment__details" do %>
+    <%= tag.h2 class: "gem-c-attachment__title" do %>
+      <%= link_to attachment.title, attachment.url,
+            class: "govuk-link",
+            target: target %>
+    <% end %>
 
-  <% if attributes.any? %>
-    <%= tag.p raw(attributes.join(', ')), class: "gem-c-attachment__metadata" %>
-  <% end %>
+    <% if attributes.any? %>
+      <%= tag.p raw(attributes.join(', ')), class: "gem-c-attachment__metadata" %>
+    <% end %>
+
+    <% unless hide_help_text %>
+      <% if attachment.opendocument? %>
+        <%= tag.p class: "gem-c-attachment__metadata" do %>
+          <%= t("components.attachment.opendocument_html", target: target) %>
+        <% end %>
+      <% end %>
+    <% end %>
 
-  <% unless hide_help_text %>
-    <% if attachment.opendocument? %>
-      <%= tag.p class: "gem-c-attachment__metadata" do %>
-        <%= t("components.attachment.opendocument_html", target: target) %>
+    <% if contact_email %>
+      <%= tag.p t("components.attachment.request_format_text"), class: "gem-c-attachment__metadata" %>
+      <%= render "govuk_publishing_components/components/details", {
+        title: t("components.attachment.request_format_cta")
+      } do %>
+        <%= t("components.attachment.request_format_details_html", contact_email: contact_email) %>
```

Comment:
This should be inside the help text conditional - as that should include all supplementary data beyond the file name, content_type, size and number of pages.

Side note I don't think help_text is the best name for that block so any suggestions on how we can either name it better or document it better are welcomed.

Reply:
It might help to know that we're only going to pass the `contact_email` attribute from Content Publisher when we actually want to render it. Not really keen on the 'pass all the data and then hide it' approach, as it seems pretty coupled to our use case. Keeping the conditional just around the open document rendering might also make it easier to rename, perhaps to a more positive 'show open document' kind of attribute?

Reply:
For govspeak it's necessary to use the all data approach as we don't know if it'll be used by an Attachment or an AttachmentLink. So there is already precedent for it. I can understand your concern, but I find it easier to think of the attachment as a model of data not a means of options.

With regards to not passing in contact_email in some cases, that seems a reversal of the precedents set so far where we have one method used both attachments and attachments link. So it'd be good if didn't intentionally veer into inconsistency without consideration.

The expectation of the email is that if one is not provided it falls back to a default one - not sure if we'll provide that or govspeak but that's what was previously part of govspeak.

On a side note, positive attributes - like you're suggestion of show - seem to not be at all common in govuk_publishing_components. Possibly due to the more verbose syntax needed than the or equal operator. When Emma and I looked into it there only seemed to be hide ones so did not want to introduce an inconsistency.

Reply:
So I think we're both agreeing on the 'model of data' approach, but there's already quite a precedent for only rendering certain things if certain pieces of that data are present.

The default email is also something that I felt was more a concern of our app than a rendering issue, as it seems more like a content issue than a rendering one. Since we have to do some presence checks in our app anyway, it seemed better to have all the logic in one place.

I found a way to modify the helper stuff we have to both simplify them and only pass the contact email where necessary. Admittedly, I did do this on the assumption we would be doing all the decision logic in our app. I'll try and get a PR up so you can see the two together...

Reply:
Just writing up here: we discussed this in-person and agreed to continue as-is, after considering the pros and cons of each approach.